x-pack/plugins/fleet/public/types/ui_extensions.ts

```diff
@@ -97,8 +97,30 @@ export interface PackageCustomExtension {
   component: LazyExoticComponent<PackageCustomExtensionComponent>;
 }
 
+/**
+ * UI Component Extension for displaying custom views under the Assets tab for a given Integration
+ */
+export type PackageAssetsComponent = ComponentType<{}>;
+
+/** Extension point registration contract for Integration details Assets view */
+export interface PackageAssetsExtension {
+  package: string;
+  view: 'package-detail-assets';
+  component: LazyExoticComponent<PackageAssetsComponent>;
+}
+
+export type AgentEnrollmentFlyoutFinalStepExtensionComponent = ComponentType<{}>;
+
+export interface AgentEnrollmentFlyoutFinalStepExtension {
+  package: string;
+  view: 'agent-enrollment-flyout';
+  component: LazyExoticComponent<AgentEnrollmentFlyoutFinalStepExtensionComponent>;
```

Comment:
what if you adjusted this interface to accept a `title` field that you can then use to modify the step title?

Reply:
This definitely makes sense, then the apps that want to hook into this functionality will simply need to export something like `stepProps` in the extension data to control what we pass into `EuiSteps`. 

The next question I have here is how to handle this change in contract in our `useUiExtensionHook`, which currently expects there to be a `component` in each of these extension interfaces:

https://github.com/elastic/kibana/blob/de1979570bd5f78255d28efaf743180d96f39172/x-pack/plugins/fleet/public/hooks/use_ui_extension.ts#L20-L37

Reply:
I think the solution I'm going to work towards will be to simply include a new hook `useAgentEnrollmentFlyoutExtension` to capture this special case for now.

Reply:
@kpollich I think we can have a `component` and a `title` in our `AgentEnrollmentFlyoutFinalStepExtension` type and you can build the step later when you retrieve the extension with that, I am wondering if the `useUIExtension` should return the whole extension instead of just the component too.

Reply:
I just pushed up https://github.com/elastic/kibana/pull/103554/commits/e4fb321fa76ec69f427de29ba52c881a1b2676ff which adds a separate hook that does what you're mentioning. We could instead refactor `useUIExtension` to return the entire extension object and then update all our usages of `useUIExtensiion` (only a few places) to do something like 

```tsx
const extension = useUIExtension(...)

// ...

if (extension) { 
  <extension.Component />
}
```

Is that what you're thinking here?

Reply:
Yes it was I was thinking,  and in the case of the step do something like this

```
const extension = useUIExtension(...)

// ...


if (extension) { 
const step = { title: extension.title, children: (<extension.Component />)}

}
```